python/pyarrow/scalar.pxi

```diff
@@ -586,6 +586,57 @@ cdef class ListValue(ArrayValue):
         return result
 
 
+cdef class LargeListValue(ArrayValue):
```

Comment:
It's not possible to have this subclass ListValue to reduce code duplication? (getitem, iter, as_py look the same)

Reply:
The problem is that the C++ types are different (e.g. `ListArray` vs. `LargeListArray`), and those need to be compile-time constants for Cython.

Reply:
Code like this makes me wish for some kind of macro system in Cython.